qdrant/utils.go

```diff
@@ -2,55 +2,47 @@ package qdrant
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
-	"net/http"
 	"strings"
 	"time"
 
 	"github.com/google/uuid"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-
-	qc "github.com/qdrant/terraform-provider-qdrant-cloud/v1/internal/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 const (
 	requestIDHeaderField = "x-qd-request-id"
 )
 
-// getErrorMessage Fetches a human readable error message out of the provided HTTP Response.
-func getErrorMessage(message string, response *http.Response) string {
-	if response == nil {
-		return "No response"
-	}
-	return fmt.Sprintf("%s: [%s: %d] - %s", message, getRequestID(response), response.StatusCode, response.Status)
-}
-
 // getRequestID fetches the Request ID from the current HTTP Response (or an empty string if not available).
-func getRequestID(response *http.Response) string {
-	if response == nil {
-		return "No response"
-	}
-	return response.Header.Get(requestIDHeaderField)
+func getRequestID(header metadata.MD) string {
+	return fmt.Sprintf(" [%s]", strings.Join(header.Get(requestIDHeaderField), "-"))
```

Comment:
I don't think this function should return a formatted string. Also, should it handle the case when the header isn't present? If that isn't possible, we should update the comment.

Reply:
An empty string is now returned, and added 'humanized' to the comments, so we can use it to log (not for inspection the log by code)